### PR TITLE
Improve robustness of heatmap test.

### PR DESCRIPTION
**Pull request by manthey**
The animation frame was mocked after an animation frame might have been requested, which could lead to a failure to step through animations properly.

### CODE REVIEW

File: tests/cases/heatmap.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Test geo.core.osmLayer
+// Test geo.core.heatmap
```
Comment (aashish24) [inline tests/cases/heatmap.js]: good catch

File: tests/cases/heatmap.js [hunk 2]
```diff
@@ -31,6 +31,7 @@ describe('canvas heatmap feature', function () {
   });
 
   it('Setup map', function () {
+    mockAnimationFrame();
```
Comment (aashish24) [inline tests/cases/heatmap.js]: why this change (moving the code to here)? I guess I have an idea but still curious

**Comment by manthey** [comment]
We need to run `mockAnimationFrame` before there is a chance that something calls `requestAnimationFrame`.  `map.draw` can do that, and this was being mocked after that.

**Comment by aashish24** [comment]
> We need to run mockAnimationFrame before there is a chance that something calls requestAnimationFrame. map.draw can do that, and this was being mocked after that.

Okay thanks not sure how to test the failed case but the explanation makes sense to me. 

LGTM :+1:
